opentreemap/stormwater/tests.py

```diff
@@ -85,36 +95,58 @@ def _get_runoff_reduced(self, feature, expect_empty=False):
             self.assert_basis(basis, 1, 0)
             return benefits['resource']['stormwater']['value']
 
-    def _assert_runoff_reduced(self, area_sq_meters, diversion_rate,
-                               runoff_reduced):
+    def _assert_runoff_reduced(self, area_sq_meters, drainage_area_sq_meters,
+                               diversion_rate, runoff_reduced):
         area = area_sq_meters * FEET_SQ_PER_METER_SQ
+        drainage_area = drainage_area_sq_meters * FEET_SQ_PER_METER_SQ
         rainfall_ft = self.instance.annual_rainfall_inches * FEET_PER_INCH
-        expected = rainfall_ft * area * diversion_rate * GALLONS_PER_CUBIC_FT
+        adjusted_area = area + (drainage_area * diversion_rate)
+        expected = rainfall_ft * adjusted_area * GALLONS_PER_CUBIC_FT
         self.assertAlmostEqual(runoff_reduced, expected, places=0)
 
     def test_rain_garden(self):
         RainGarden.set_config_property(self.instance, 'should_show_eco', True)
-        feature = self._make_map_feature(RainGarden)
+        drainage_area_sq_meters = 100000000.0
+        feature = self._make_map_feature(RainGarden, drainage_area_sq_meters)
         runoff_reduced = self._get_runoff_reduced(feature)
         self._assert_runoff_reduced(
-            self.polygon_area_sq_meters, .95, runoff_reduced)
+            self.polygon_area_sq_meters, drainage_area_sq_meters,
+            .85, runoff_reduced)
 
     def test_bioswale(self):
-        feature = self._make_map_feature(Bioswale)
+        drainage_area_sq_meters = 100000000.0
+        feature = self._make_map_feature(Bioswale, drainage_area_sq_meters)
         runoff_reduced = self._get_runoff_reduced(feature)
         # Note Bioswale diversion rate set to .5 in setUp()
         self._assert_runoff_reduced(
-            self.polygon_area_sq_meters, .5, runoff_reduced)
+            self.polygon_area_sq_meters, drainage_area_sq_meters,
+            .5, runoff_reduced)
+
+    def test_bioswale_no_drainage(self):
+        feature = self._make_map_feature(Bioswale)
+        runoff_reduced = self._get_runoff_reduced(feature, expect_empty=True)
+        self.assertEqual(runoff_reduced, 0)
 
     def test_eco_not_wanted(self):
         RainGarden.set_config_property(self.instance, 'should_show_eco', False)
-        feature = self._make_map_feature(RainGarden)
+        drainage_area_sq_meters = 100000000.0
+        feature = self._make_map_feature(RainGarden,
+                                         drainage_area=drainage_area_sq_meters)
         runoff_reduced = self._get_runoff_reduced(feature, expect_empty=True)
         self.assertEqual(runoff_reduced, 0)
 
     def test_bulk(self):
-        self._make_map_feature(Bioswale)
-        self._make_map_feature(Bioswale)
+        drainage_area_sq_meters = 100000000.0
+        # NOTE
+        # Today, no check is made for overlapping stormwater resources.
+        # In the future, either overlap should be invalid,
```

Comment:
I would like to open an issue to implement a validation that prevents creating overlapping polygonal map features.